glean-core/python/glean/glean.py

```diff
@@ -213,6 +214,9 @@ def set_upload_enabled():
 
                 if original_enabled is False and cls.get_upload_enabled() is True:
                     cls._initialize_core_metrics()
+                else:
```

Comment:
Should the logic here be `if original_enabled is True and cls.get_upload_enabled() is False` (so that it only happens the first time on toggling, not if `False` is passed twice?

Reply:
Actually ... I think so, which means we need to change that on the Kotlin/Swift side as well